dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java

```diff
@@ -342,20 +336,49 @@ private Object getRelationshipInfo(final Field field)
 		final ContentletRelationshipRecords records = relationships.getRelationshipsRecords().get(0);
 		if(records.getRecords().isEmpty()) {
 		    return null;
+		} else if (records.doesAllowOnlyOne()){
+			return new ContentMap(records.getRecords().get(0),user, EDIT_OR_PREVIEW_MODE, host, context);
+		} else {
+			final List<Contentlet> contents = records.getRecords().stream()
+					.filter(contentlet -> needLiveVersion(contentlet))
+					.map(relatedContent -> EDIT_OR_PREVIEW_MODE ? relatedContent : getLiveVersion(relatedContent))
+					.collect(Collectors.toList());
+
+			return perAPI.filterCollection(contents, PermissionAPI.PERMISSION_USE, true, user).stream()
+					.map(contentlet -> new ContentMap(contentlet, user, EDIT_OR_PREVIEW_MODE, host, context))
+					.collect(Collectors.toList());
 		}
-		else if (records.doesAllowOnlyOne()){
-			return new ContentMap(records.getRecords().get(0),user,
-					EDIT_OR_PREVIEW_MODE, host, context);
-		} else{
-			return perAPI.filterCollection(records.getRecords(),
-					PermissionAPI.PERMISSION_USE, true, user).stream()
-					.map(contentlet -> new ContentMap((Contentlet) contentlet, user,
-							EDIT_OR_PREVIEW_MODE, host, context)).collect(Collectors.toList());
-		}
 
 
 	}
 
+	private Contentlet getLiveVersion(final Contentlet relatedContent) {
+		try {
+			if (relatedContent.isLive()) {
+				return relatedContent;
+			} else {
+				try {
+					final List<Contentlet>  contentlets = APILocator.getContentletAPI()
+							.findContentletsByIdentifiers(new String[]{relatedContent.getIdentifier()}, true, relatedContent.getLanguageId(), user, user.isFrontendUser());
+
+					return contentlets.get(0);
+				} catch (DotDataException| DotSecurityException e) {
+					throw new RuntimeException(e);
+				}
+			}
+		} catch (DotDataException | DotSecurityException e) {
+			return relatedContent;
+		}
+	}
+
+	private boolean needLiveVersion(final Contentlet contentlet) {
+		try {
+			return EDIT_OR_PREVIEW_MODE || !EDIT_OR_PREVIEW_MODE && contentlet.hasLiveVersion();
```

Comment:
you ca use Try.of

Reply:
done https://github.com/dotCMS/core/pull/19551/commits/9ed56d60337cd0089bc8c6b1562ecf5fcd027a50#diff-fbb11290830742365067c6c4e761a23f3eef90d1e024037cbaa272100c134e61R376